### PR TITLE
Feature/online state

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

File: plugins/fbs/fbs.js [hunk 1]
```diff
@@ -314,7 +314,88 @@ FBS.prototype.block = function block(username, reason) {
...
+            }
+
+            // Send state event into the bus.
+            var eventName = onlineState.online ? 'fbs.online' : 'fbs.offline';
+            bus.emit(eventName, {
+              timestamp: new Date().getTime(),
+              online: onlineState
+            });
+
+            console.log('', onlineState);
```
Comment (tuj) [inline plugins/fbs/fbs.js]: console.log?

File: plugins/fbs/fbs.js [hunk 2]
```diff
@@ -314,7 +314,88 @@ FBS.prototype.block = function block(username, reason) {
...
+            // Error connecting to FBS.
+            onlineState.online = false;
+            onlineState.currentHold = 0;
+            setTimeout(checkOnlineState, onlineState.offlineTimeout);
+            bus.emit(fbs.offline, {
+              timestamp: new Date().getTime(),
+              online: onlineState
+            });
+
+            console.log('', onlineState);
```
Comment (tuj) [inline plugins/fbs/fbs.js]: console.log?

File: public/js/services/configService.js [hunk 3]
```diff
@@ -79,19 +79,19 @@ angular.module('BibBox').service('configService', ['$rootScope', '$translate', '
...
-  default_lang: 'da',
-  initialized: false,
-  debug: false,
-  translations: {},
-  languages: [],
-  features: [],
-  timeout: {
-    idleTimeout: 15,
-    IdleWarn: 5
+  "default_lang": 'da',
```
Comment (tuj) [inline public/js/services/configService.js]: Without ""

File: plugins/fbs/fbs.js [hunk 4]
```diff
@@ -314,7 +314,88 @@ FBS.prototype.block = function block(username, reason) {
 module.exports = function (options, imports, register) {
   var bus = imports.bus;
 
-  // Create FBS object to use in tests.
+  // Defines the configuration for the online checker below.
+  var onlineState = {
+    'online': true,
```
Comment (tuj) [inline plugins/fbs/fbs.js]: without ''

File: plugins/fbs/fbs.js [hunk 5]
```diff
@@ -314,7 +314,88 @@ FBS.prototype.block = function block(username, reason) {
...
+    'threshold': 5,
+    'currentHold': 5,
```
Comment (tuj) [inline plugins/fbs/fbs.js]: What is the meaning of currentHold. Perhaps another word could make med understand it :)
Perhaps
  successfulOnlineChecks

File: public/js/controllers/indexController.js [hunk 6]
```diff
@@ -64,18 +64,9 @@ angular.module('BibBox').controller('IndexController', ['$rootScope', '$scope',
...
-     *        front-end has the information and can react on it if need be!
      */
     var fbsOnline = function () {
-      userService.isOnline().then(function (status) {
-        $scope.online = status;
-      },
-      function (err) {
-        $scope.online = false;
-      });
+      $scope.online = userService.isOnline();
```
Comment (tuj) [inline public/js/controllers/indexController.js:69]: What about when state changes while indexController is running, and has already called isOnline() ?

File: public/js/services/userService.js [hunk 7]
```diff
@@ -304,29 +304,34 @@ angular.module('BibBox').service('userService', ['$q', '$timeout', '$location',
     /**
      * Check FBS is online.
      *
-     * It's plased on user service as FBS has to due with users.
+     * It's placed on user service as FBS has to due with users.
```
Comment (tuj) [inline public/js/services/userService.js]: "This function is placed on user service as all FBS calls have to do with users."

**Review by tuj — CHANGES_REQUESTED** [review]

File: public/js/controllers/indexController.js (hunk 6 above)
Comment (cableman) [reply public/js/controllers/indexController.js:69]: It still has it's old online interval checker.

https://github.com/bibboxen/bibbox/blob/38dcea72a05c40ec8f648aa5de7a269d84e9c48a/public/js/controllers/indexController.js#L77

File: plugins/fbs/fbs.js (hunk 4 above)
Comment (cableman) [reply plugins/fbs/fbs.js]: Removed '

File: plugins/fbs/fbs.js (hunk 5 above)
Comment (cableman) [reply plugins/fbs/fbs.js]: Renamed

File: plugins/fbs/fbs.js (hunk 1 above)
Comment (cableman) [reply plugins/fbs/fbs.js]: UPS... removed

File: plugins/fbs/fbs.js (hunk 2 above)
Comment (cableman) [reply plugins/fbs/fbs.js]: UPS... removed

File: public/js/services/configService.js (hunk 3 above)
Comment (cableman) [reply public/js/services/configService.js]: Removed "

File: public/js/services/userService.js (hunk 7 above)
Comment (cableman) [reply public/js/services/userService.js]: Copyed in

**Comment by cableman** [comment]
Review changes added to the code base.

**Review by tuj — APPROVED** [review]

**Pull request merged**